logstash-core/src/main/java/org/logstash/Event.java

```diff
@@ -264,16 +264,16 @@ public Event clone() throws CloneNotSupportedException {
     }
 
     public String toString() {
-        // TODO: (colin) clean this IOException handling, not sure why we bubble IOException here
-        try {
-            return (getTimestamp().toIso8601() + " " + this.sprintf("%{host} %{message}"));
-        } catch (IOException e) {
-            String host = (String)this.data.get("host");
-            host = (host != null ? host : "%{host}");
+        Object f = this.getField("host");
+        String host = (f != null ? f.toString() : "%{host}");
 
-            String message = (String)this.data.get("message");
-            message = (message != null ? message : "%{message}");
+        f = this.getField("message");
+        String message = (f != null ? f.toString() : "%{message}");
 
+        try {
+            // getTimestamp throws an IOException if there is no @timestamp field
+            return (getTimestamp().toIso8601() + " " + host + " " + message);
```

Comment:
exactly. that is why I wrote this is the PR description. 

> Will also followup with an issue to review the Event.getTimestamp() behaviour.

but changing `getTimestamp()` this is out of scope for the bugfix & PR.